### PR TITLE
Iwyu

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
It's all about build time, pruning the includes.
I was driving myself crazy with one-off include pruning and realized I was wasting my time.

#### Describe the solution
Do some runs of IWYU and then do some manual fixups for the places where it went crazy.
I did one slightly more involved change, which was factoring the avatar.h dependency out of talker_avatar.h by pushing the pointer handling down into the compilation unit instead of the header. This isn't going to have any visible impact as the reason this was compiling before was avatar.h was included in every compilation unit before talker_avatar.h, but this makes things slightly more robust.

#### Describe alternatives you've considered
It will probably be productive to follow up with the latest IWYU release, but that was more time than I was willing to put into it, this is actually the clang_7.0 branch of IWYU.
IWYU inexplicably moved around some includes, which I'm not a huge fan of, but this was already a huge time sink so I didn't feel like fixing it right now, I'll probably come back and address that later.

#### Testing
It builds and stuff, I'm interested to see what CBA has to say about it.
This has a decent chance of breaking in sdltiles.cpp/h or something.

### CODE REVIEW

File: src/catacharset.cpp [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
-#include <array>
 
 #include "options.h"
 #include "output.h"
 #include "wcwidth.h"
 
 #if defined(_WIN32)
-#include "platform_win.h"
 #include "mmsystem.h"
+#include "platform_win.h"
```
Comment (ZhilkinSerg) [inline src/catacharset.cpp:14]:
```suggestion
```

File: src/catacharset.cpp [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 #include <cstdlib>
 #include <cstring>
-#include <array>
 
 #include "options.h"
 #include "output.h"
 #include "wcwidth.h"
 
 #if defined(_WIN32)
-#include "platform_win.h"
```
Comment (ZhilkinSerg) [inline src/catacharset.cpp:12]:
```suggestion
#include "platform_win.h"
```

File: src/catacharset.cpp [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+#include <array>
 #include <cstdlib>
 #include <cstring>
-#include <array>
 
 #include "options.h"
 #include "output.h"
 #include "wcwidth.h"
 
 #if defined(_WIN32)
```
Comment (ZhilkinSerg) [inline src/catacharset.cpp:12]:
```suggestion
#if defined(_WIN32)
#include "platform_win.h"
```
```

Comment (ZhilkinSerg) [reply src/catacharset.cpp:12]: Should not be reordered alphabetically and `platform_win.h` should be included before `mmsystem.h`.

File: src/debug_menu.cpp [hunk 4]
```diff
@@ -1,58 +1,56 @@
 #include "debug_menu.h"
 
+
```
Comment (ZhilkinSerg) [inline src/debug_menu.cpp:3]:
```suggestion
```

File: src/debug_menu.cpp [hunk 5]
```diff
@@ -1,58 +1,56 @@
...
 #include "coordinates.h"
-#include "cursesdef.h"
+#include "creature.h"
 #include "debug.h"
 #include "dialogue_chatbin.h"
 #include "enum_conversions.h"
 #include "enums.h"
 #include "event.h"
 #include "event_bus.h"
 #include "faction.h"
```
Comment (ZhilkinSerg) [inline src/debug_menu.cpp:43]:
```suggestion
#include "faction.h"
#include "filesystem.h"
```

File: src/debug_menu.cpp [hunk 6]
```diff
@@ -1,58 +1,56 @@
...
-#include "filesystem.h"
```
Comment (ZhilkinSerg) [inline src/debug_menu.cpp:47]: It is required for code guarded by `TILES` define (for `assure_dir_exist`, etc).

File: src/input.cpp [hunk 7]
```diff
@@ -26,14 +26,12 @@
 #include "options.h"
 #include "output.h"
 #include "path_info.h"
+#include "point.h"
 #include "popup.h"
 #include "string_formatter.h"
 #include "string_input_popup.h"
 #include "translations.h"
 #include "ui_manager.h"
```
Comment (ZhilkinSerg) [inline src/input.cpp:34]: It is required for code guarded by TILES define (`get_window_dimensions`, etc). Maybe we should guard this include too or move called function to a different header?

```suggestion
#include "ui_manager.h"
#include "sdltiles.h"
```

File: src/init.cpp [hunk 8]
```diff
@@ -74,11 +71,11 @@
 #include "recipe_dictionary.h"
 #include "recipe_groups.h"
 #include "regional_settings.h"
+#include "relic.h"
 #include "requirements.h"
 #include "rotatable_symbols.h"
 #include "scenario.h"
 #include "scent_map.h"
```
Comment (ZhilkinSerg) [inline src/init.cpp:78]: It is required for code guarded by TILES.

```suggestion
#include "scent_map.h"
#include "sdltiles.h"
```

File: src/game.cpp [hunk 9]
```diff
@@ -134,7 +134,6 @@
 #include "scenario.h"
 #include "scent_map.h"
 #include "scores_ui.h"
```
Comment (ZhilkinSerg) [inline src/game.cpp:136]:
```suggestion
#include "scores_ui.h"
#include "sdltiles.h"
```

File: src/game.cpp [hunk 10]
```diff
@@ -54,7 +52,7 @@
 #include "coordinate_conversions.h"
 #include "coordinates.h"
 #include "creature_tracker.h"
```
Comment (ZhilkinSerg) [inline src/game.cpp:54]:
```suggestion
#include "creature_tracker.h"
#include "cursesport.h"
```

File: src/debug_menu.cpp [hunk 11]
```diff
@@ -75,15 +73,12 @@
 #include "overmap.h"
 #include "overmap_ui.h"
 #include "overmapbuffer.h"
```
Comment (ZhilkinSerg) [inline src/debug_menu.cpp:75]:
```suggestion
#include "overmapbuffer.h"
#include "path_info.h"
```

File: src/output.cpp [hunk 12]
```diff
@@ -1,41 +1,39 @@
...
-#include <array>
-#include <memory>
 #include <type_traits>
-#include <cmath>
+#include <vector>
 
 #include "cata_utility.h"
 #include "catacharset.h"
 #include "color.h"
 #include "cursesdef.h"
```
Comment (ZhilkinSerg) [inline src/output.cpp:24]:
```suggestion
#include "cursesdef.h"
#include "cursesport.h"
```

File: src/options.cpp [hunk 13]
```diff
@@ -1,29 +1,30 @@
...
 #include "game_constants.h"
 #include "input.h"
 #include "json.h"
+#include "line.h"
 #include "mapsharing.h"
 #include "output.h"
 #include "path_info.h"
 #include "point.h"
 #include "popup.h"
 #include "sdlsound.h"
```
Comment (ZhilkinSerg) [inline src/options.cpp:27]:
```suggestion
#include "sdlsound.h"
#include "sdltiles.h"
```

File: src/options.cpp [hunk 14]
```diff
@@ -1,29 +1,30 @@
...
 #include <climits>
 #include <iterator>
 #include <stdexcept>
+#include <type_traits>
 
 #include "calendar.h"
 #include "cata_utility.h"
 #include "catacharset.h"
 #include "color.h"
 #include "cursesdef.h"
```
Comment (ZhilkinSerg) [inline src/options.cpp:14]:
```suggestion
#include "cursesdef.h"
#include "cursesport.h"
```

File: src/sounds.cpp [hunk 15]
```diff
@@ -40,12 +33,8 @@
 #include "string_id.h"
 #include "translations.h"
 #include "type_id.h"
```
Comment (ZhilkinSerg) [inline src/sounds.cpp:35]:
```suggestion
#include "type_id.h"
#include "vpart_position.h"
```

File: src/ui_manager.cpp [hunk 16]
```diff
@@ -8,7 +8,6 @@
 #include "cursesdef.h"
 #include "game_ui.h"
 #include "point.h"
```
Comment (ZhilkinSerg) [inline src/ui_manager.cpp:10]:
```suggestion
#include "point.h"
#include "sdltiles.h"
```

File: src/weather_gen.cpp [hunk 17]
```diff
@@ -1,20 +1,21 @@
...
-#include <chrono>
 #include <cmath>
 #include <ostream>
 #include <random>
 #include <string>
+#include <utility>
 
 #include "cata_utility.h"
 #include "game_constants.h"
 #include "json.h"
```
Comment (ZhilkinSerg) [inline src/weather_gen.cpp:12]: This one is probably mandatory for Windows-builds, so we can guard it with ifdef.
```suggestion
#include "json.h"
#include "math_defines.h"
```

File: src/sounds.cpp [hunk 18]
```diff
@@ -5,31 +5,24 @@
...
 #include "calendar.h"
 #include "character.h"
 #include "coordinate_conversions.h"
-#include "creature.h"
+#include "coordinates.h"
 #include "debug.h"
 #include "effect.h"
 #include "enums.h"
 #include "game.h"
 #include "game_constants.h"
```
Comment (ZhilkinSerg) [inline src/sounds.cpp:18]:
```suggestion
#include "game_constants.h"
#include "itype.h"
```

File: src/animation.cpp [hunk 19]
```diff
@@ -1,24 +1,29 @@
...
+#include "input.h"
 #include "map.h"
+#include "memory_fast.h"
 #include "monster.h"
 #include "mtype.h"
 #include "options.h"
 #include "output.h"
 #include "player.h"
+#include "point.h"
 #include "popup.h"
```
Comment (ZhilkinSerg) [inline src/animation.cpp:21]: Required for MSVC builds.
```suggestion
#include "popup.h"
#include "posix_time.h"
```

File: src/iuse_software_kitten.cpp [hunk 20]
```diff
@@ -1,11 +1,13 @@
 #include "iuse_software_kitten.h"
 
+#include <time.h>
+#include <algorithm>
 #include <cstdlib>  // Needed for rand()
 #include <vector>
 
+#include "cuboid_rectangle.h"
 #include "input.h"
 #include "output.h"
```
Comment (ZhilkinSerg) [inline src/iuse_software_kitten.cpp:10]:
```suggestion
#include "output.h"
#include "posix_time.h"
```

File: src/debug.cpp [hunk 21]
```diff
@@ -10,15 +10,13 @@
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
```
Comment (ZhilkinSerg) [inline src/debug.cpp:12]:
```suggestion
#include <ctime>
#include <fstream>
```

File: src/debug.cpp [hunk 22]
```diff
@@ -10,15 +10,13 @@
...
-#include <fstream>
 #include <iomanip>
 #include <iterator>
 #include <locale>
 #include <map>
 #include <memory>
 #include <set>
```
Comment (ZhilkinSerg) [inline src/debug.cpp:18]:
```suggestion
#include <set>
#include <sstream>
```

File: src/item_location.cpp [hunk 23]
```diff
@@ -1,6 +1,9 @@
 #include "item_location.h"
 
+#include <stddef.h>
```
Comment (kevingranade) [inline src/item_location.cpp:3]:
```suggestion
#include <cstddef>
```

File: src/item_pocket.h [hunk 24]
```diff
@@ -2,30 +2,38 @@
 #ifndef CATA_SRC_ITEM_POCKET_H
 #define CATA_SRC_ITEM_POCKET_H
 
+#include <stddef.h>
```
Comment (kevingranade) [inline src/item_pocket.h:5]:
```suggestion
#include <cstddef>
```

File: src/magic_teleporter_list.cpp [hunk 25]
```diff
@@ -1,27 +1,27 @@
 #include "magic_teleporter_list.h"
 
+#include <stddef.h>
```
Comment (kevingranade) [inline src/magic_teleporter_list.cpp:3]:
```suggestion
#include <cstddef>
```

File: src/magic_teleporter_list.cpp [hunk 26]
```diff
@@ -1,27 +1,27 @@
...
 #include <algorithm>
```
Comment (kevingranade) [inline src/magic_teleporter_list.cpp:4]:
```suggestion
#include <algorithm>
#include <cstddef>
```

File: src/move_mode.cpp [hunk 27]
```diff
@@ -1,7 +1,14 @@
 #include "move_mode.h"
 
+#include <stddef.h>
+#include <algorithm>
```
Comment (kevingranade) [inline src/move_mode.cpp:4]:
```suggestion
#include <algorithm>
#include <cstddef>
```

File: src/proficiency.cpp [hunk 28]
```diff
@@ -1,5 +1,10 @@
 #include "proficiency.h"
 
+#include <stddef.h>
+#include <algorithm>
```
Comment (kevingranade) [inline src/proficiency.cpp:4]:
```suggestion
#include <algorithm>
#include <cstddef>
```

File: src/relic.cpp [hunk 29]
```diff
@@ -1,14 +1,18 @@
 #include "relic.h"
 
+#include <stdlib.h>
 #include <algorithm>
 #include <cmath>
```
Comment (kevingranade) [inline src/relic.cpp:5]:
```suggestion
#include <algorithm>
#include <cmath>
#include <cstdlib>
```

File: src/requirements.cpp [hunk 30]
```diff
@@ -1,5 +1,6 @@
 #include "requirements.h"
 
+#include <math.h>
 #include <algorithm>
 #include <cassert>
 #include <climits>
```
Comment (kevingranade) [inline src/requirements.cpp:6]:
```suggestion
#include <algorithm>
#include <cassert>
#include <climits>
#include <cmath>
```

File: src/rotatable_symbols.cpp [hunk 31]
```diff
@@ -1,15 +1,16 @@
 #include "rotatable_symbols.h"
 
-#include <cstdlib>
-#include <array>
-#include <vector>
+#include <math.h>
 #include <algorithm>
+#include <array>
 #include <cstdint>
+#include <cstdlib>
```
Comment (kevingranade) [inline src/rotatable_symbols.cpp:7]:
```suggestion
#include <algorithm>
#include <array>
#include <cstdint>
#include <cstdlib>
#include <cmath>
```

File: src/savegame_json.cpp [hunk 32]
```diff
@@ -1,15 +1,12 @@
...
 // includes in such headers.
-#include "enums.h" // IWYU pragma: associated
-#include "npc_favor.h" // IWYU pragma: associated
-#include "pldata.h" // IWYU pragma: associated
 
+#include <math.h>
 #include <algorithm>
 #include <array>
 #include <bitset>
 #include <climits>
```
Comment (kevingranade) [inline src/savegame_json.cpp:9]:
```suggestion
#include <algorithm>
#include <array>
#include <bitset>
#include <climits>
#include <cmath>
```

File: src/scenario.cpp [hunk 33]
```diff
@@ -1,7 +1,8 @@
 #include "scenario.h"
 
-#include <cstdlib>
+#include <math.h>
 #include <algorithm>
```
Comment (kevingranade) [inline src/scenario.cpp:4]:
```suggestion
#include <algorithm>
#include <cmath>
```

File: src/scent_map.cpp [hunk 34]
```diff
@@ -1,5 +1,6 @@
 #include "scent_map.h"
 
+#include <math.h>
 #include <algorithm>
 #include <cassert>
 #include <cstdlib>
```
Comment (kevingranade) [inline src/scent_map.cpp:6]:
```suggestion
#include <algorithm>
#include <cassert>
#include <cstdlib>
#include <cmath>
```

File: src/translations.h [hunk 35]
```diff
@@ -2,12 +2,14 @@
 #ifndef CATA_SRC_TRANSLATIONS_H
 #define CATA_SRC_TRANSLATIONS_H
 
+#include <stddef.h>
```
Comment (kevingranade) [inline src/translations.h:5]:
```suggestion
#include <cstddef>
```

File: src/vehicle_display.cpp [hunk 36]
```diff
@@ -1,9 +1,8 @@
-#include "vehicle.h" // IWYU pragma: associated
-
-#include <cstdlib>
+#include <math.h>
 #include <algorithm>
```
Comment (kevingranade) [inline src/vehicle_display.cpp:2]:
```suggestion
#include <algorithm>
#include <cmath>
```

File: src/weather_type.cpp [hunk 37]
```diff
@@ -1,6 +1,10 @@
 #include "weather_type.h"
-#include "weather.h"
-#include "game_constants.h"
+
+#include <stdlib.h>
```
Comment (kevingranade) [inline src/weather_type.cpp:3]:
```suggestion
#include <cstdlib>
```

File: src/achievement.cpp [hunk 38]
```diff
@@ -1,9 +1,22 @@
 #include "achievement.h"
 
+#include <assert.h>
```
Comment (kevingranade) [inline src/achievement.cpp:3]:
```suggestion
#include <cassert>
```

File: src/achievement.cpp [hunk 39]
```diff
@@ -1,9 +1,22 @@
...
+#include <stdlib.h>
```
Comment (kevingranade) [inline src/achievement.cpp:4]:
```suggestion
#include <cstdlib>
```

File: src/activity_actor.cpp [hunk 40]
```diff
@@ -1,29 +1,38 @@
 #include "activity_actor.h"
 
+#include <math.h>
+#include <array>
```
Comment (kevingranade) [inline src/activity_actor.cpp:4]:
```suggestion
#include <array>
#include <cmath>
```

File: src/animation.cpp [hunk 41]
```diff
@@ -1,24 +1,30 @@
 #include "animation.h"
 
+#include <time.h>
```
Comment (kevingranade) [inline src/animation.cpp:3]:
```suggestion
#include <ctime>
```

File: src/avatar_action.cpp [hunk 42]
```diff
@@ -1,5 +1,6 @@
 #include "avatar_action.h"
 
+#include <math.h>
 #include <algorithm>
 #include <climits>
```
Comment (kevingranade) [inline src/avatar_action.cpp:5]:
```suggestion
#include <algorithm>
#include <climits>
#include <cmath>
```

File: src/computer_session.cpp [hunk 43]
```diff
@@ -1,18 +1,20 @@
 #include "computer_session.h"
 
+#include <math.h>
 #include <algorithm>
```
Comment (kevingranade) [inline src/computer_session.cpp:4]:
```suggestion
#include <algorithm>
#include <cmath>
```

File: src/consumption.cpp [hunk 44]
```diff
@@ -1,5 +1,4 @@
-#include "player.h" // IWYU pragma: associated
-
+#include <math.h>
 #include <algorithm>
 #include <array>
```
Comment (kevingranade) [inline src/consumption.cpp:3]:
```suggestion
#include <algorithm>
#include <array>
#include <cmath>
```

File: src/craft_command.cpp [hunk 45]
```diff
@@ -1,11 +1,13 @@
 #include "craft_command.h"
 
+#include <math.h>
 #include <algorithm>
 #include <climits>
```
Comment (kevingranade) [inline src/craft_command.cpp:5]:
```suggestion
#include <algorithm>
#include <climits>
#include <cmath>
```

File: src/creature.cpp [hunk 46]
```diff
@@ -1,5 +1,6 @@
 #include "creature.h"
 
+#include <math.h>
 #include <algorithm>
 #include <array>
```
Comment (kevingranade) [inline src/creature.cpp:5]:
```suggestion
#include <algorithm>
#include <array>
#include <cmath>
```

File: src/grab.cpp [hunk 47]
```diff
@@ -1,19 +1,20 @@
-#include "game.h" // IWYU pragma: associated
-
-#include <cstdlib>
+#include <math.h>
 #include <algorithm>
```
Comment (kevingranade) [inline src/grab.cpp:2]:
```suggestion
#include <algorithm>
#include <cmath>
```

File: src/handle_action.cpp [hunk 48]
```diff
@@ -1,13 +1,13 @@
-#include "game.h" // IWYU pragma: associated
-
+#include <math.h>
+#include <algorithm>
 #include <chrono>
```
Comment (kevingranade) [inline src/handle_action.cpp:3]:
```suggestion
#include <algorithm>
#include <chrono>
#include <cmath>
```

File: src/item_pocket.cpp [hunk 49]
```diff
@@ -1,20 +1,32 @@
 #include "item_pocket.h"
 
+#include <math.h>
+#include <stdlib.h>
+#include <algorithm>
```
Comment (kevingranade) [inline src/item_pocket.cpp:5]:
```suggestion
#include <algorithm>
#include <cmath>
#include <cstdlib>
```

File: src/iuse_software_kitten.cpp [hunk 50]
```diff
@@ -1,8 +1,11 @@
 #include "iuse_software_kitten.h"
 
+#include <time.h>
+#include <algorithm>
 #include <cstdlib>  // Needed for rand()
```
Comment (kevingranade) [inline src/iuse_software_kitten.cpp:5]:
```suggestion
#include <algorithm>
#include <cstdlib>  // Needed for rand()
#include <ctime>
```

File: src/map_extras.cpp [hunk 51]
```diff
@@ -1,5 +1,7 @@
 #include "map_extras.h"
 
+#include <math.h>
+#include <algorithm>
 #include <array>
```
Comment (kevingranade) [inline src/map_extras.cpp:5]:
```suggestion
#include <algorithm>
#include <array>
#include <cmath>
```

File: src/map_field.cpp [hunk 52]
```diff
@@ -1,3 +1,4 @@
+#include <math.h>
 #include <algorithm>
 #include <array>
 #include <bitset>
```
Comment (kevingranade) [inline src/map_field.cpp:4]:
```suggestion
#include <algorithm>
#include <array>
#include <bitset>
#include <cmath>
```

File: src/mapgen_functions.cpp [hunk 53]
```diff
@@ -1,5 +1,6 @@
 #include "mapgen_functions.h"
 
+#include <math.h>
 #include <algorithm>
 #include <array>
```
Comment (kevingranade) [inline src/mapgen_functions.cpp:5]:
```suggestion
#include <algorithm>
#include <array>
#include <cmath>
```

File: src/martialarts.cpp [hunk 54]
```diff
@@ -1,13 +1,16 @@
 #include "martialarts.h"
 
+#include <math.h>
 #include <algorithm>
```
Comment (kevingranade) [inline src/martialarts.cpp:4]:
```suggestion
#include <algorithm>
#include <cmath>
```

File: src/mission_companion.cpp [hunk 55]
```diff
@@ -1,25 +1,26 @@
 #include "mission_companion.h"
 
+#include <math.h>
```
Comment (kevingranade) [inline src/mission_companion.cpp:3]:
```suggestion
```

File: src/mission_companion.cpp [hunk 56]
```diff
@@ -1,25 +1,26 @@
...
 #include <algorithm>
-#include <array>
 #include <cassert>
```
Comment (kevingranade) [inline src/mission_companion.cpp:5]:
```suggestion
#include <cassert>
#include <cmath>
```

File: src/mutation.h [hunk 57]
```diff
@@ -2,7 +2,10 @@
 #ifndef CATA_SRC_MUTATION_H
 #define CATA_SRC_MUTATION_H
 
+#include <limits.h>
+#include <algorithm>
```
Comment (kevingranade) [inline src/mutation.h:6]:
```suggestion
#include <algorithm>
#include <climits>
```

File: src/mutation_data.cpp [hunk 58]
```diff
@@ -1,17 +1,18 @@
-#include "mutation.h" // IWYU pragma: associated
-
-#include <array>
+#include <stdlib.h>
```
Comment (kevingranade) [inline src/mutation_data.cpp:1]:
```suggestion
#include <cstdlib>
```

File: src/ncurses_def.cpp [hunk 59]
```diff
@@ -14,11 +16,14 @@
 #endif
 
 #include <langinfo.h>
+#include <string.h>
```
Comment (kevingranade) [inline src/ncurses_def.cpp:19]:
```suggestion
```

File: src/newcharacter.cpp [hunk 60]
```diff
@@ -1,7 +1,5 @@
-#include "avatar.h" // IWYU pragma: associated
-
+#include <math.h>
```
Comment (kevingranade) [inline src/newcharacter.cpp:1]:
```suggestion
```

File: src/newcharacter.cpp [hunk 61]
```diff
@@ -1,7 +1,5 @@
...
 #include <algorithm>
-#include <array>
 #include <climits>
```
Comment (kevingranade) [inline src/newcharacter.cpp:3]:
```suggestion
#include <climits>
#include <cmath>
```

File: src/npctrade.cpp [hunk 62]
```diff
@@ -1,16 +1,17 @@
 #include "npctrade.h"
 
+#include <math.h>
 #include <algorithm>
```
Comment (kevingranade) [inline src/npctrade.cpp:4]:
```suggestion
#include <algorithm>
#include <cmath>
```

File: src/pathfinding.cpp [hunk 63]
```diff
@@ -1,28 +1,29 @@
 #include "pathfinding.h"
 
-#include <cstdlib>
+#include <math.h>
```
Comment (kevingranade) [inline src/pathfinding.cpp:3]:
```suggestion
```

File: src/pathfinding.cpp [hunk 64]
```diff
@@ -1,28 +1,29 @@
...
 #include <algorithm>
-#include <queue>
-#include <set>
 #include <array>
```
Comment (kevingranade) [inline src/pathfinding.cpp:5]:
```suggestion
#include <array>
#include <cmath>
```

File: src/player_display.cpp [hunk 65]
```diff
@@ -1,5 +1,4 @@
-#include "player.h" // IWYU pragma: associated
-
+#include <math.h>
 #include <algorithm>
 #include <array>
```
Comment (kevingranade) [inline src/player_display.cpp:3]:
```suggestion
#include <algorithm>
#include <array>
#include <cmath>
```

File: src/relic.h [hunk 66]
```diff
@@ -2,22 +2,30 @@
 #ifndef CATA_SRC_RELIC_H
 #define CATA_SRC_RELIC_H
 
+#include <limits.h>
+#include <math.h>
+#include <algorithm>
```
Comment (kevingranade) [inline src/relic.h:7]:
```suggestion
#include <algorithm>
#include <climits>
#include <cmath>
```

File: src/vehicle_move.cpp [hunk 67]
```diff
@@ -1,40 +1,41 @@
-#include "vehicle.h" // IWYU pragma: associated
-
+#include <assert.h>
 #include <algorithm>
 #include <array>
```
Comment (kevingranade) [inline src/vehicle_move.cpp:3]:
```suggestion
#include <algorithm>
#include <array>
#include <cassert>
```

File: src/avatar.h [hunk 68]
```diff
@@ -3,32 +3,45 @@
...
+class item_location;
 class mission;
 class monster;
+class nc_color;
 class npc;
 class talker;
+namespace catacurses
+{
+class window;
+}
```
Comment (ZhilkinSerg) [inline src/avatar.h:43]:
```suggestion
} // namespace catacurses
```

File: src/options.cpp [hunk 69]
```diff
@@ -1,9 +1,11 @@
 #include "options.h"
 
+#include <locale.h>
```
Comment (ZhilkinSerg) [inline src/options.cpp:3]:
```suggestion
#include <clocale>
```

**Pull request merged**